kernel/aster-nix/src/net/iface/common.rs

```diff
@@ -147,56 +149,93 @@ impl IfaceCommon {
 
     pub(super) fn poll<D: Device + ?Sized>(&self, device: &mut D) {
         let mut interface = self.interface.lock_irq_disabled();
+        let mut sockets = self.sockets.lock_irq_disabled();
+
```

Comment:
The lock order matters to avoid ABBA deadlock. So we always follows the same order to acquire multiple locks.

In current code base, it seems we acquire the lock on `sockets` first, then `interface`( line55 and 68 in [process_dhcp](
https://github.com/asterinas/asterinas/blob/e83e1fc01ba38ad2a405d7d710ec7258fb664f60/kernel/aster-nix/src/net/iface/virtio.rs#L54), line126 and 129 in [do_connect](https://github.com/asterinas/asterinas/blob/e83e1fc01ba38ad2a405d7d710ec7258fb664f60/kernel/aster-nix/src/net/iface/any_socket.rs#L125)).

So it's better to follow the similar lock order

Reply:
The problems described in this comment and https://github.com/asterinas/asterinas/pull/1098#discussion_r1692435152 exist before this PR. I have made two new commits to fix the behavior. Thanks for pointing this out!